Bump closedxml to v.0.100.3

vbondarev opened the pull request:
I found a new static method [FromObject](https://github.com/ClosedXML/ClosedXML/blob/dce92f82300293c874014c33e9d8be181a717707/ClosedXML/Excel/XLCellValue.cs#L193) on the XLCellValue type and copied it. In the future, when a new version of ClosedXML is released, we can replace `XLCellValueConverter.FromObject` with `XLCellValue.FromObject`.

PR is in the draft because some unit tests are fail. The problem is that ClosedXml reads empty сells as a text, but ClosedXml.Report as Blank.Value. For example:  `Customers(templateFile: "GroupTagTests_DisableOutline.xlsx")` returns
```
Xunit.Sdk.XunitException
Expected areEqual to be true because Cell data types are not equal starting from C1, but found False.
   at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
   at FluentAssertions.Execution.AssertionScope.FailWith(Func`1 failReasonFunc)
   at FluentAssertions.Primitives.BooleanAssertions`1.BeTrue(String because, Object[] becauseArgs)
   at ClosedXML.Report.Tests.XlsxTemplateTestsBase.CompareWithGauge(XLWorkbook actual, String fileExpected) in D:\Work\projects\ClosedXML.Report\tests\ClosedXML.Report.Tests\XlsxTemplateTestsBase.cs:line 77
   at ClosedXML.Report.Tests.GroupTagTests.<>c__DisplayClass3_0.<Customers>b__1(XLWorkbook wb) in D:\Work\projects\ClosedXML.Report\tests\ClosedXML.Report.Tests\GroupTagTests.cs:line 96
   at ClosedXML.Report.Tests.XlsxTemplateTestsBase.XlTemplateTest(String tmplFileName, Action`1 arrangeCallback, Action`1 assertCallback) in D:\Work\projects\ClosedXML.Report\tests\ClosedXML.Report.Tests\XlsxTemplateTestsBase.cs:line 45
   at ClosedXML.Report.Tests.GroupTagTests.Customers(String templateFile) in D:\Work\projects\ClosedXML.Report\tests\ClosedXML.Report.Tests\GroupTagTests.cs:line 85
``` 

![image](https://user-images.githubusercontent.com/3437698/229138002-0193b8f6-4f24-4062-9566-b5a761e76b64.png)
![image](https://user-images.githubusercontent.com/3437698/229139077-add00682-3987-471a-bd46-93f47bae30e6.png)
![image](https://user-images.githubusercontent.com/3437698/229139260-6b8d3b75-87a2-49d0-b9be-50871bfadfc2.png)

If anyone has any ideas, welcome)

vbondarev committed:
#303 Updated dependencies

vbondarev committed:
#303: Bump to ClosedXML 0.100.3 (unit tests fail)

Pankraty commented:
@vbondarev, the enum `XLDataType` has been modified in the version 0.100: https://github.com/ClosedXML/ClosedXML/pull/1922/files#diff-7adf695e4309b9c5f943cfffbff6ff3e24809a5bb3157cc15cc9028e07d41008R19

Previously, there was no such thing as a "blank value", empty cells were treated as if they stored empty strings.

In order to make the tests passing, I would suggest doing this:

1. Temporarily disable the comparison of data types and make sure all tests are green.
2. Visually verify that the produced files (from `.\tests\ClosedXML.Report.Tests\bin\Debug\net6.0\Output`) look the same as the reference files (from `.\tests\Gauges`).
3. Replace the reference files with the generated ones and enable the comparison back. After this, all tests should be green again.

Thank you so much for contributing!

Pankraty committed:
Fix assigning formulas in group totals

Pankraty committed:
Disable test for formula with tags; fix template for not using formulas with tags inside

Upgrading to ClosedXML 0.100 broke this functionality: previously it was possible to put a formula inside a cell and then to retrieve an original value it had (e.g. &="Total:"<<sum>>). Now this does not work (which is correct) and making it right would require somewhat major refactoring for which no one has time

Pankraty committed:
Fix cell value conversion to account for the values already represented as XLCellValue

Pankraty committed:
Tweak cell data type comparison in tests

Pankraty commented:
Believe it or not, but I finally managed to upgrade C.Report to ClosedXML 0.100.3 🎉 

@vbondarev many thanks for your effort!
